AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -703,6 +704,89 @@ - (instancetype)styledWithBlock:(void (^)(ASLayoutElementStyle *style))styleBloc
 
 #pragma mark - Layout
 
+- (void)invalidateSize
+{
+  ASDisplayNodeAssertThreadAffinity(self);
+  
+  __instanceLock__.lock();
+
+  // Mark the node for layout in the next layout pass
+  [self invalidateCalculatedLayout];
+  
+  if (_supernode) {
+    ASDisplayNode *supernode = _supernode;
+    __instanceLock__.unlock();
+    // Cause supernode's layout to be invalidated
+    // We need to release the lock to prevent a deadlock
+    [supernode invalidateSize];
+    return;
+  }
+  
+  // Calculate a new pending layout. It will be applied in the next layout pass
+  ASLayout *layout = [self layoutThatFits:_calculatedDisplayNodeLayout->constrainedSize];
+  if (CGSizeEqualToSize(self.bounds.size, layout.size) == NO) {
+    // If the size of the layout changes inform the node of this
```

Comment:
Inform our container (e.g. table, collection) that we need it to change our bounds size.